m24sr_driver.cpp

```diff
@@ -193,15 +193,15 @@ static M24srError_t is_correct_crc_residue(uint8_t *data, uint8_t length) {
             return M24SR_IO_ERROR_CRC;
         } else {
             /* Good CRC, but error status from M24SR */
-            status = (M24srError_t) (((data[1] << 8) & 0xFF00) | (data[2] & 0x00FF));
+            status = ((data[1] << 8) & 0xFF00) | (data[2] & 0x00FF);
         }
     }
 
     if (status == NFC_COMMAND_SUCCESS) {
-        status = M24SR_SUCCESS;
+        return M24SR_SUCCESS;
     }
 
-    return status;
+    return M24SR_ERROR;
```

Comment:
this now makes all errors look the same

Reply:
@paul-szczepanek-arm 
changed to `return static_cast<M24srError_t>(status);`